src/proposal.mligo

```diff
@@ -231,14 +272,16 @@ let unfreeze_proposer_and_voter_token
             in
       (tokens, desired_burn_amount)
     in
+
+  // Do the unstake for the proposer
   let store = unstake_tk(tokens, burn_amount, proposal.proposer, period, store) in
 
-  // unfreeze_voter_token
-  let do_unfreeze = fun
-        ( store, voter_with_vote : storage * ((address * vote_type) * nat))
-          -> unstake_tk(voter_with_vote.1, 0n, voter_with_vote.0.0, period, store) in
+  // Update `is_flushed_or_dropped`
+  let new_proposals = Map.add proposal_key
+        ({proposal with is_flushed_or_dropped = true}) store.proposals in
```

Comment:
Isn't it odd that this happens in a function called `unstake_proposer_token`?

I mean, by the name they don't seem related

Reply:
Since I've removed `is_flushed_or_dropped`, I also this part as well.